src/dropdown/Dropdown.js

```diff
@@ -0,0 +1,188 @@
+  let { indexSelected } = props
+  const defaultSelected = data[0]
+  const label = typeof defaultSelected.label !== 'undefined' ? defaultSelected.label : defaultSelected.value
+  const [isListShown, setIsListShown] = React.useState(false)
+  const [dataDropdown, setDataDropdown] = React.useState([])
+  const tmpItemSelected = typeof itemSelected === 'undefined' ? null : itemSelected
+  const [selectedItem, setSelectedItem] = React.useState(tmpItemSelected)
+  const input = React.useRef(null)
+  const [inputHeigh, setInputHeigh] = React.useState(0)
+  const [index, setIndex] = React.useState(indexSelected)
+
+  const _onFocus = () => {
+    console.log("_onFocus")
+
+    if (isListShown) {
+      setIsListShown(false)
+    } else {
+      setIsListShown(true)
+    }
+  }
+
+  React.useEffect(() => {
+    setDataDropdown(data)
+  }, [])
+
+  React.useEffect(() => {
+    console.log("item selected")
+    // setSelectedItem(itemSelected)
+    if (typeof itemSelected !== 'undefined') {
+      dataDropdown.map((item, index) => {
+        if (itemSelected.value === item.value) {
+          setSelectedItem(item)
+          indexSelected = index
+        }
+      })
+    }
+  }, [itemSelected])
+
+  React.useEffect(() => {
+    console.log("index selected")
+    // setSelectedItem(itemSelected)
+    if (typeof indexSelected !== 'undefined') {
+      dataDropdown.map((item, index) => {
+        if (indexSelected === index) {
+          setSelectedItem(item)
+        }
+      })
+    }
+  }, [indexSelected])
+
+  const _selectThisItem = (selected) => {
+    console.log("selected: ", selected)
+    setIsListShown(false)
+    setSelectedItem(selected)
+    input.current.blur()
+    onItemSelected(selected)
+  }
+
+  return (
+    <View style={[{
+      width: '90%',
+      margin: 4,
+      zIndex: 1000,
+      backgroundColor: 'white',
+      borderBottomWidth: 1,
+      borderBottomColor: 'rgba(0,0,0,.2)'
+    }, containerStyle]}
+      onLayout={(event) => {
+        var { x, y, width, height } = event.nativeEvent.layout;
+        setInputHeigh(height)
+      }}
+    >
+      <TextInput
+        style={[{
+            fontSize: 17,
+            padding: 8
+          }, innerStyle]}
+        ref={input}
+        value={selectedItem !== null ? (selectedItem.label ? selectedItem.label : selectedItem.value) : label}
+        rightIcon={{ type: 'font-awesome', name: 'chevron-down' }}
+        editable={false}
+        onTouchStart={() => {
+          console.log("onFocus");
+          _onFocus()
+        }}
+        onBlur={() => {
+          console.log("onBlur")
+        }}
+
+      />
+      {isListShown ? (
+        <ListDropDown isListShown data={dataDropdown} selectThisItem={_selectThisItem}
+          inputHeigh={inputHeigh} selectedItem={selectedItem} containerListStyle listStyle />
+      ) : (
+          <View />
+        )}
+    </View>
+
+  )
+}
+
+const ListDropDown = (props) => {
```

Comment:
You could just deconstruct the props here
```suggestion
const ListDropDown = ({ data, selectThisItem, inputHeigh, containerListStyle, listStyle }) => {
```